src/rust/core/server/src/threads/listener.rs

```diff
@@ -38,7 +38,7 @@ pub struct Listener {
     poll: Poll,
     session_queue: QueuePairs<Session, ()>,
     ssl_context: Option<SslContext>,
-    signal_queue: QueuePairs<(), Signal>,
+    signal_queue: QueuePairs<Signal, Signal>,
```

Comment:
In places where we have both directions utilized, it'll be good to add comment on what the Signal is intended (nature of the message) when they are declared, since the signature is symmetrical and abstract.

Reply:
See if the comments in the lastest commit help here or if there's something else you're looking for.